packages/library-base/src/commonMain/kotlin/io/realm/kotlin/RealmConfiguration.kt

```diff
@@ -128,8 +128,16 @@ public interface RealmConfiguration : Configuration {
                 schema,
                 LogConfiguration(logLevel, allLoggers),
                 maxNumberOfActiveVersions,
-                notificationDispatcher ?: singleThreadDispatcher(fileName),
-                writeDispatcher ?: singleThreadDispatcher(fileName),
+                if (notificationDispatcher != null) {
```

Comment:
I think you could remove the null assertion and delete the if with:

`notificationDispatcher?.let { CoroutineDispatcherFactory.external(it) }`

and

`writeDispatcher?.let { CoroutineDispatcherFactory.external(it) }`

then make make the dispatchers constructor parameters in `RealmConfigurationImpl` nullable, to finally 
change the parameters in the `ConfigurationImpl` constructor to

`notificationDispatcher ?: CoroutineDispatcherFactory.internal(name)` 

🤯

Reply:
True, but that would also split the logic where we set up the RealmConfiguration. I agree the above code is a bit messy. I have tried to move it around a little to hopefully make it more clear.

Reply:
Yes, I came to say that you could save the `let` by setting the value in the builder.